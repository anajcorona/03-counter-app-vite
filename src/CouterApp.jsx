// cuando cambia el estado se vuelve a renderear

import PropTypes from 'prop-types';
import { useState } from 'react';
export const CounterApp = ({ value }) => {
    
    const  [ counter, setCounter ] = useState( value );
    const handleAdd = (e) => {
        setCounter( counter + 1)
        // Se puede usar de etsa manera
        // setCounter( ()=> 2000)
    }
    const handleLess = () => setCounter( counter - 1);
    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            
            {/* Se puede usar solo el handleAdd y se pasa el eveto principal porque solo tenemos un parametro 
            Si se requieren mas parametros ya de coloca como ...handledAdd(event, param2, param3)  */}
            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleLess }> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
        </>
    )
}
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}