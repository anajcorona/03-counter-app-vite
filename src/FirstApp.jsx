// //import { Fragment } from "react";
// //el Fragmemnt es igual al <> </>
// //<Fragment> </Fragment>
// const newMessage = {
//   name: 'Ana',
//   age: 26
// };

// function hello() {
//     return 'Wenas Wenas!';
// }

// // const lol = () => {
// //   return (
// //     <div>HelloWorldApp</div>
// //     )
// //   }
  
  
//   export const FirstApp = () => {
  
//     //Puede ir adentro o afuera dependiendo del scoope XD si queda afuera es maa efectivo por la memoria
//     // const HelloWorldApp = () => {
//     //   return 'Holaa'
//     // }

//     return (
//       <>
//         <h1> { JSON.stringify(newMessage) } </h1>
//         <h3> { hello() + ' ' + newMessage.name } </h3>
//         {/* <h3> { lol() } </h3>
//         <h3> { HelloWorldApp() } </h3> */}
//         {/* <code> { JSON.stringify(newMessage) } </code> */}
//         <p> Lalala </p>
//       </>
//     )
//   }

import PropTypes from 'prop-types';
  
export const FirstApp = ( { title, subtitle, name } ) => {
  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      <h2> { subtitle } </h2>
      <h2> { subtitle } </h2>

      <h2> { name } </h2>


      <p> Lalala </p>
    </>
  )
}

  //Define si una propiedas es requerida y de que valor debe ser
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

// define los valores por default de las propiedades
// Se ejecutan primero que los propTypes
FirstApp.defaultProps = {
//  title: "No hay titutlo",
  subtitle: "No hay subtitile",
  name: "Ana"
}
  