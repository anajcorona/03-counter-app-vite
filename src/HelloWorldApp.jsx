// export function App() {
//     return (<h1> Holiiiiiiiiii !!!!</h1>);
// }

export const HelloWorldApp = () => {
  return (
    <div>HelloWorldApp</div>
  )
}

//comando rapido rafc, te crea la segunda parte mas completa

// si exportas por default el nombre por default en el import es el que quieras ejemplo:
// export default App
// import Lala from './Hellos...'

