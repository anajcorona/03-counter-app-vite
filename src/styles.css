html, body {
    background-color: black;
    color: aquamarine;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    background-color:bisque;
    padding: 1rem;
    margin: 0.5rem;
    border: 2px solid palevioletred;
}