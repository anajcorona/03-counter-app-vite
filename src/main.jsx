import React from 'react';
import ReactDOM from 'react-dom/client';
//import { FirstApp } from './FirstApp';
//import { HelloWorldApp } from './HelloWorldApp';
import './styles.css';
import {CounterApp} from './CouterApp';

ReactDOM.createRoot( document.getElementById("root")).render(    
    <React.StrictMode>
        <CounterApp  value={ 26 }/>
        {/* <FirstApp title='Hola soy Ana'/> */}
        {/* <HelloWorldApp /> */}
    </React.StrictMode>
);
