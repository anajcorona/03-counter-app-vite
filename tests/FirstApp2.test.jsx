import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas <FirstApp />', () => {

    const title = 'Hola, Soy Goku';
    const subtitle = 'Hay un subtitle';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title = { title } />);
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar ele mensaje "Hola, soy Goku"', () => {
        //screen.debug();
        render(<FirstApp title = { title } />);
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe mostrar el titulo en un H1', () => {
        render(<FirstApp title = { title } />);
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain(title);
    });

    test('Debe mostrar el Subtitulo enviado por props', () => {
        render(
            <FirstApp 
                subtitle = { subtitle }    
                title = { title } 
            />
        );

        expect( screen.getAllByText(subtitle).length).toBe(2);
    });
});