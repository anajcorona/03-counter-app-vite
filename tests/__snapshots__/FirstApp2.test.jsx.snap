// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <h2>
     
    No hay subtitile
     
  </h2>
  <h2>
     
    No hay subtitile
     
  </h2>
  <h2>
     
    Ana
     
  </h2>
  <p>
     Lalala 
  </p>
</div>
`;
